src/platform/Darwin/DnssdInterfaceInfos.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <dns_sd.h>
+#include <lib/dnssd/platform/Dnssd.h>
+
+#include <string>
+#include <vector>
+
+namespace chip {
+namespace Dnssd {
+
+struct RegisterContext;
+struct RegisterRecordContext;
+
+class InterfaceInfos
+{
+public:
+    bool Init(const struct ifaddrs * ifa, Inet::IPAddressType addressType, uint32_t interfaceId);
```

Comment:
Please document what a false return value means.  In particular, it might just be that this ifaddrs is not allowed by the addressType/interfaceId we pass in.